src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriter.cs

```diff
@@ -207,25 +216,42 @@ protected override void WriteStartChangesetImplementation(string changeSetId)
         /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
         /// <param name="payloadUriOption">
         /// The format of operation Request-URI, which could be AbsoluteUri, AbsoluteResourcePathAndHost, or RelativeResourcePath.</param>
-        /// <param name="dependsOnIds">The prerequisite request ids of this request.</param>
+        /// <param name="dependsOnIds">The prerequisite request ids of this request. It should be null for Multipart/Mixed format
+        /// and, therefore, validation will pass automatically.</param>
         /// <returns>The message that can be used to write the request operation.</returns>
         protected override ODataBatchOperationRequestMessage CreateOperationRequestMessageImplementation(
             string method, Uri uri, string contentId, BatchPayloadUriOption payloadUriOption,
             IEnumerable<string> dependsOnIds)
         {
+            if (dependsOnIds != null)
+            {
+                throw new ODataException(
+                    "For Multipart/Mixed batch format, request dependency is implicit by the protocol and should be null.");
+            }
+
             // write pending message data (headers, response line) for a previously unclosed message/request
             this.WritePendingMessageData(true);
 
             // create the new request operation
+            // For Multipart batch format, dependsOnIds is implicitly derived from batch message itself and is not
+            // user input, therefore no validation is needed.
             ODataBatchOperationRequestMessage operationRequestMessage = BuildOperationRequestMessage(
                 this.RawOutputContext.OutputStream,
-                method, uri, contentId, /*groupId*/null, dependsOnIds, ODataFormat.Batch);
+                method, uri, contentId,
+                ODataMultipartMixedBatchWriterUtils.GetChangeSetIdFromBoundary(this.changeSetBoundary),
+                this.dependsOnIdsTracker.GetDependsOnIds(),
+                /*dependsOnIdsValidationRequired*/ false);

```

Comment:
>/*dependsOnIdsValidationRequired*/ false [](start = 16, length = 40)

Not sure why we wouldn't pass true in order to validate against the ids already read? Is this just for backward compatibility? Seems like the request is destined to fail if it doesn't reference a valid id.

Reply:
Updated to use only dependsOnIds which is only null for the implicit dependsOnIds Multipart/Mixed case; and the GetDependsOnIds overrides in derived classes getting the convertedDependsOnIds in the BuildOperationRequestMessage method.

---
In reply to: [158351447](https://github.com/OData/odata.net/pull/1020#discussion_r158351447) [](ancestors = 158351447)